lib/chef/provisioning/transport/ssh.rb

```diff
@@ -38,6 +41,7 @@ def initialize(host, username, ssh_options, options, global_config)
         @username = username
         @ssh_options = ssh_options
         @options = options
+        @scp_temp_dir = options.fetch(:scp_temp_dir, '/tmp')
```

Comment:
@tyler-ball @glennmatthews I think there may be issue with this line.

My co-worker @andrewjamesbrown was hitting error which latest chefdk 0.7.0 that I was unable to produce on 0.6.2. It seems when you have have specified machine_options attribute in machine resource block then @options becomes a Cheffish::MergedConfig object. 

https://github.com/chef/chef-provisioning/blob/master/lib/chef/resource/machine.rb#L101

This object has not implemented the fetch method so we're seeing an exception.